src/main/java/org/kohsuke/github/GHCheckRun.java

```diff
@@ -72,11 +103,106 @@ public String getHeadSha() {
     }
 
     /**
-     * @deprecated This object has no HTML URL.
+     * Gets the HTML URL: https://github.com/[owner]/[repo-name]/runs/[check-run-id], usually an GitHub Action page of
+     * the check run.
+     *
+     * @return HTML URL
      */
     @Override
     public URL getHtmlUrl() {
-        return null;
+        return htmlUrl;
+    }
+
+    /**
+     * Gets the global node id to access most objects in GitHub.
+     *
+     * @see <a href="https://developer.github.com/v4/guides/using-global-node-ids/">documentation</a>
+     * @return Global node id
+     */
+    public String getNodeId() {
+        return nodeId;
+    }
+
+    /**
+     * Gets a reference for the check run on the integrator's system.
+     *
+     * @return Reference id
+     */
+    public String getExternalId() {
+        return externalId;
+    }
+
+    /**
+     * Gets the details URL from which to find full details of the check run on the integrator's site.
+     *
+     * @return Details URL
+     */
+    public URL getDetailsUrl() {
+        return detailsUrl;
+    }
+
+    /**
+     * Gets the start time of the check run in ISO 8601 format: YYYY-MM-DDTHH:MM:SSZ.
+     *
+     * @return Timestamp of the start time
+     */
+    public Date getStartedAt() {
+        return GitHubClient.parseDate(startedAt);
+    }
+
+    /**
+     * Gets the completed time of the check run in ISO 8601 format: YYYY-MM-DDTHH:MM:SSZ.
+     *
+     * @return Timestamp of the completed time
+     */
+    public Date getCompletedAt() {
+        return GitHubClient.parseDate(completedAt);
+    }
+
+    /**
+     * Gets the GitHub app this check run belongs to, included in response.
+     *
+     * @return GitHub App
+     */
+    public GHApp getApp() {
+        return app;
+    }
+
+    public Output getOutput() {
+        return output;
+    }
+
+    /**
+     * Represents an output in a check run to include summary and other results.
+     *
+     * @see <a href="https://developer.github.com/v3/checks/runs/#output-object">documentation</a>
+     */
+    public static class Output {
+        private String title;
+        private String summary;
+        private String text;
+        private int annotationsCount;
+        private URL annotationsUrl;
```

Comment:
Do we have a `CheckRunAnnotations` yet? If not we'll need that added eventually as well.

Reply:
That's where I'm confused. The response (or event payload) from GitHub does not include `annotation` objects, but they are needed when users are making requests. So does our implementation have to make sure it supports both usages?

Reply:
Most probably it should support. Would be beneficial for users in the longer term

Reply:
It isn't required for this PR, I'm just noting it for later.